blender/arm/material/make_depth.py

```diff
@@ -46,11 +76,24 @@ def make(context_id, rpasses, shadowmap=False):
     vert.write_attrib('vec4 spos = vec4(pos.xyz, 1.0);')
     vert.add_include('compiled.inc')
 
-    parse_opacity = 'translucent' in rpasses or mat_state.material.arm_discard or 'refraction' in rpasses
+    parse_opacity = 'translucent' in rpasses or 'refraction' in rpasses or mat_state.material.arm_discard
 
     parse_custom_particle = (cycles.node_by_name(mat_state.nodes, 'ArmCustomParticleNode') is not None)
 
-    if parse_opacity:
+    if shadowmap_transparent:
+        billboard = mat_state.material.arm_billboard
+        if billboard == 'spherical':
+            vert.add_uniform('mat3 N', '_normalMatrixSphere')
+        elif billboard == 'cylindrical':
+            vert.add_uniform('mat3 N', '_normalMatrixCylinder')
+        else:
+            vert.add_uniform('mat3 N', '_normalMatrix')
+        make_mesh._write_material_attribs_default(frag, parse_opacity) #TODO remove duplicate parsing
+        vert.add_out('vec3 wnormal')
+        make_attrib.write_norpos(con_depth, vert)
+        frag.write_attrib('vec3 n = normalize(wnormal);')
+        cycles.parse(mat_state.nodes, con_depth, vert, frag, geom, tesc, tese, basecol_only=True, parse_opacity=True)
```

Comment:
We need basecol, which sometimes need some nodes inputs values. If we didn't need a special vertex shader setup we'd just call make_forward_base.